prediction-applier/src/main/java/org/alfresco/hxi_connector/prediction_applier/auth/PredictionApplierHxAuthClient.java

```diff
@@ -51,4 +65,44 @@ public AuthenticationResult authenticate(String tokenUri, ClientRegistration cli
     {
         return super.authenticate(tokenUri, clientRegistration);
     }
+
+    @Retryable(retryFor = EndpointServerErrorException.class,
+            maxAttemptsExpression = "#{@hxInsightProperties.hylandExperience.authentication.retry.attempts}",
+            backoff = @Backoff(delayExpression = "#{@hxInsightProperties.hylandExperience.authentication.retry.initialDelay}",
+                    multiplierExpression = "#{@hxInsightProperties.hylandExperience.authentication.retry.delayMultiplier}"))
+    @Override
+    public AuthenticationResult authenticate(String clientRegistrationId)
+    {
+        OAuth2ClientProperties.Provider provider = oAuth2ClientProperties.getProvider().get(clientRegistrationId);
+        Objects.requireNonNull(provider, "Auth Provider not found for client registration id: " + clientRegistrationId);
+        OAuth2ClientProperties.Registration registration = oAuth2ClientProperties.getRegistration().get(clientRegistrationId);
+        Objects.requireNonNull(registration, "Auth Registration not found for client registration id: " + clientRegistrationId);
+        String tokenUri = provider.getTokenUri();
+        ClientRegistration clientRegistration = ClientRegistration.withRegistrationId(clientRegistrationId)
+                .tokenUri(tokenUri)
+                .authorizationGrantType(new AuthorizationGrantType(registration.getAuthorizationGrantType()))
+                .clientId(registration.getClientId())
+                .clientSecret(registration.getClientSecret())
+                .scope(registration.getScope())
+                .build();
+        return super.authenticate(tokenUri, clientRegistration);
+    }
+
+    @Override
+    protected String createEncodedBody(ClientRegistration clientRegistration)
+    {
+        if (AuthorizationGrantType.PASSWORD.getValue().equals(clientRegistration.getAuthorizationGrantType().getValue()))
```

Comment:
This is special handling of `password` grant type which is not supported by Spring Boot OAuth2 auto configuration.